docs/agent-oriented-approach.md

```diff
@@ -0,0 +1,49 @@
+# Agent-oriented problem solving
```

Comment:
Can we use `agent-oriented` throughout rather than `agent-based`? I want to avoid confusion/conflation with agent based modelling

Reply:
I can try to change it in as many places as possible, but some aren't quite correct when agent-oriented is used.